General.ruleset

```diff
@@ -0,0 +1,119 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```

Comment:
There's a curious thing:

This code trips [SA1116](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1116.md), rightly so:

```
public Task UnverifiedUserShouldBeRedirectedToEmailConfirmationPageAfterLogin(Browser browser) =>
    ExecuteTestAfterSetup(context =>
    {
        context.SignUp();

        context.LogOff();

        context.LogIn(OrdinaryUser.Email, OrdinaryUser.Password);
        AssertEmailConfirmationPage(context);
    }, browser);
```

See `EmailConfirmationTests` in the Finitve solution (or almost any other UI test class there).

However, this doesn't cause a warning:

```
public override IDisplayResult Edit(DocumentPart part, BuildPartEditorContext context) =>
    Initialize<EditDocumentPartViewModel>($"{nameof(DocumentPart)}_Edit", model =>
    {
        model.EndDateUtc = part.EndDateUtc;
        model.InternalNotes = part.InternalNotes;
        model.PrimaryContactEmail = part.PrimaryContactEmail;
        model.SecondaryContactEmail = part.SecondaryContactEmail;
        model.SignatureState = part.SignatureState;
        model.StartDateUtc = part.StartDateUtc;
        model.TermLengthMonths = part.TermLengthMonhts;
        model.TermType = part.TermType;
        model.ExternalPath = part.ExternalPath;
    }).Location("Content: 10");
```
See `DocumentPartDisplayDriver` (or a lot of other *Driver classes).

Why is this?

Reply:
Strange. I have no idea. Nothing is perfect.

Reply:
Well, I know :) but do you have any suggestions on how to catch the formatting issue in the second case too? Any other rules?

Reply:
I don't know. You can, for example, raise an issue on StyleCop repository regarding that.

Reply:
Fair enough, will do.

Reply:
I see now that the issue is that that this is not actually covered by SA1116 but [SA1117](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1117.md). I enabled this by adding a `<Rule Id="SA1117" Action="Warning" />` line to the ruleset but it still doesn't notice `DocumentPartDisplayDriver`. I think it should provide a warning for both `Test` methods in the following classes but only works for the first one:

```
    public class Ok
    {
        public void Test() =>
            Invoked(input =>
            {
            }, "second argument");

        private void Invoked(Action<int> action, string secondParameter)
        {
        }
    }

    public class NotOk
    {
        public void Test() =>
            Invoked("first argument", input =>
            {
            });

        private void Invoked(string firstParameter, Action<int> action)
        {
        }
    }
```

Could you check please?

Reply:
The same for me. I don't know why it doesn't generate a warning. But at least it works for the first case. As for me, we can live with this.

Reply:
I noticed that SA1117 works in some places so I've opened a StyleCop issue about it: https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3183

Reply:
It turns out this is likely a StyleCop bug.

Reply:
If you happen to know an alternative rule from Sonar or FxCop (I checked but didn't find one) then please add that, otherwise, we'll leave this as such.

Reply:
I also don't know another similar rule.

Reply:
Let's leave this as such then.